### ISSUE TITLE
Rasa NLU Parse Error on upgrading to version 0.12 

### ISSUE THREAD

**Issue by souvikg10** [RasaHQ/rasa_nlu#1014]
**Rasa NLU version** (e.g. `0.12.0`):

**Used backend / pipeline** (`spacy_ner`, `spacy_sklearn`, ...):

**Operating system** (osx):

**Issue**: 'unicode' object has no attribute 'get'

**Content of configuration file** (if used & relevant):
```
language: "nl"

pipeline:
# this is using the spacy sklearn pipeline, adding duckling
# all components will use their default values
- name: "nlp_spacy"
- name: "tokenizer_spacy"
- name: "intent_featurizer_spacy"
- name: "ner_crf"
- name: "ner_synonyms"
- name: "intent_classifier_sklearn"
- name: "ner_duckling_http"
  locale: "nl"
- name: "intent_entity_featurizer_regex"
- name: "ner_spacy"
```

stack trace:
```
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-] Traceback (most recent call last):
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]   File "/app/rasa_nlu/rasa_nlu/server.py", line 232, in parse
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]     else threads.deferToThread(self.data_router.parse, data))
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]   File "/usr/local/lib/python2.7/site-packages/twisted/python/threadpool.py", line 250, in inContext
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]     result = inContext.theWork()
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]   File "/usr/local/lib/python2.7/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]     inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]   File "/usr/local/lib/python2.7/site-packages/twisted/python/context.py", line 122, in callWithContext
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]     return self.currentContext().callWithContext(ctx, func, *args, **kw)
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]   File "/usr/local/lib/python2.7/site-packages/twisted/python/context.py", line 85, in callWithContext
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]     return func(*args,**kw)
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]   File "/app/rasa_nlu/rasa_nlu/data_router.py", line 237, in parse
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]     model)
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]   File "/app/rasa_nlu/rasa_nlu/project.py", line 118, in parse
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]     interpreter = self._interpreter_for_model(model_name)
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]   File "/app/rasa_nlu/rasa_nlu/project.py", line 177, in _interpreter_for_model
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]     return Interpreter.create(metadata, self._component_builder)
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]   File "/app/rasa_nlu/rasa_nlu/model.py", line 298, in create
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]     components.validate_requirements(model_metadata.component_classes)
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]   File "/app/rasa_nlu/rasa_nlu/model.py", line 85, in component_classes
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-]     return [c.get("class") for c in self.get('pipeline', [])]
rasa-nlu_1             | 2018-04-18 13:53:15+0000 [-] AttributeError: 'unicode' object has no attribute 'get'
```

**Comment by tmbo** [comment]
What is it you do with that configuration file? Are you posting it to the server to train a model?

Not sure yet how you got to the error.

**Comment by souvikg10** [comment]
I am getting the error when I am trying to send a /parse requests. However my latest trained model was on a different version. I did not train a new model after upgrading the version. I am going to retrain a new model and check

**Comment by souvikg10** [comment]
I did check again sorry, It does work now however it expects the http_duckling_url in the config file, I have a docker-compose running and a container on rasa_nlu and I have an environment variable as 

RASA_DUCKLING_HTTP_URL=http://duckling:8000

Error on duckling it is not able to find.

but for the issue you can just add a warning in the migration guide that models needs to be retrained in order to make it run on the new version

**Comment by tmbo** [comment]
I guess we can close this then. I've added the warning to the docs and fixed the duckling issue. Build is currently running. Fixed as part of 0.12.1
